scipy/stats/_morestats.py

```diff
@@ -1309,29 +1321,40 @@ def _all(x):
         message = ("The `optimizer` argument of `boxcox_normmax` must return "
                    "an object containing the optimal `lmbda` in attribute `x`.")
         raise ValueError(message)
-    else:
-        # Test if the optimal lambda causes overflow
+    elif ymax is None or not np.isinf(ymax):  # adjust the final lambda
         x = np.asarray(x)
-        max_x = np.max(x, axis=0)
-        istransinf = np.isinf(special.boxcox(max_x, res))
-        dtype = x.dtype if np.issubdtype(x.dtype, np.floating) else np.float64
-        if np.any(istransinf):
-            warnings.warn(
-                f"The optimal lambda is {res}, but the returned lambda is "
-                f"the constrained optimum to ensure that the maximum of the "
-                f"transformed data does not cause overflow in {dtype}.",
-                stacklevel=2
-            )
-
-            # Return the constrained lambda to ensure the transformation
-            # does not cause overflow
-            ymax = np.finfo(dtype).max / 100  # 100 is the safety factor
-            constrained_res = _boxcox_inv_lmbda(max_x, ymax)
-
-            if isinstance(res, np.ndarray):
-                res[istransinf] = constrained_res
-            else:
-                res = constrained_res
+        xmax = np.max(x, axis=0)
+        ymax_res = special.boxcox(xmax, res)
+
+        if ymax is None:
+            # Test if the optimal lambda causes overflow
+            istransinf = np.isinf(ymax_res)
+            dtype = x.dtype if np.issubdtype(x.dtype, np.floating) else np.float64
+            if np.any(istransinf):
+                warnings.warn(
+                    f"The optimal lambda is {res}, but the returned lambda is "
+                    f"the constrained optimum to ensure that the maximum of the "
+                    f"transformed data does not cause overflow in {dtype}.",
+                    stacklevel=2
+                )
+
+                # Return the constrained lambda to ensure the transformation
+                # does not cause overflow
+                ymax = np.finfo(dtype).max / 100  # 100 is the safety factor
+                constrained_res = _boxcox_inv_lmbda(xmax, ymax)
+
+                if isinstance(res, np.ndarray):
+                    res[istransinf] = constrained_res
+                else:
+                    res = constrained_res
+        else:
+            islargethanymax = ymax_res > ymax
```

Comment:
```suggestion
            mask = ymax_res > ymax
```
or
```suggestion
            is_larger_than_ymax = ymax_res > ymax
```
I find that more readable.